lib/ui/painting/immutable_buffer.cc

```diff
@@ -49,6 +53,51 @@ void ImmutableBuffer::init(Dart_NativeArguments args) {
   tonic::DartInvoke(callback_handle, {Dart_TypeVoid()});
 }
 
+void ImmutableBuffer::initFromAsset(Dart_NativeArguments args) {
+  UIDartState::ThrowIfUIOperationsProhibited();
+  Dart_Handle callback_handle = Dart_GetNativeArgument(args, 2);
+  if (!Dart_IsClosure(callback_handle)) {
+    Dart_SetReturnValue(args, tonic::ToDart("Callback must be a function"));
+    return;
+  }
+  Dart_Handle asset_name_handle = Dart_GetNativeArgument(args, 1);
+  uint8_t* chars = nullptr;
+  intptr_t asset_length = 0;
+  Dart_Handle result =
+      Dart_StringToUTF8(asset_name_handle, &chars, &asset_length);
+  if (Dart_IsError(result)) {
+    Dart_SetReturnValue(args, tonic::ToDart("Asset must be valid UTF8"));
+    return;
+  }
+  Dart_Handle immutable_buffer = Dart_GetNativeArgument(args, 0);
+
+  std::string asset_name = std::string{reinterpret_cast<const char*>(chars),
+                                       static_cast<size_t>(asset_length)};
+
+  std::shared_ptr<AssetManager> asset_manager = UIDartState::Current()
+                                                    ->platform_configuration()
+                                                    ->client()
+                                                    ->GetAssetManager();
+  std::unique_ptr<fml::Mapping> data = asset_manager->GetAsMapping(asset_name);
+  if (data == nullptr) {
+    Dart_SetReturnValue(args, tonic::ToDart("Asset not found"));
+    return;
+  }
+
+  auto size = data->GetSize();
+  const void* bytes = static_cast<const void*>(data->GetMapping());
+
+#if FML_OS_ANDROID
+  auto sk_data = MakeSkDataWithCopy(bytes, size);
+#else
+  auto sk_data = SkData::MakeWithCopy(bytes, size);
+#endif  // FML_OS_ANDROID
+
```

Comment:
nit: I think you can just call `MakeSkDataWithCopy` since it's already ifdefed like this below anyway.

Reply:
oh woops, misread that - thanks!